tests/bin/eslint.js

```diff
@@ -1,15 +1,25 @@
 /**
- * @fileoverview Tests for the eslint.js executable.
+ * @fileoverview Integration tests for the eslint.js executable.
  * @author Teddy Katz
  */
 
 "use strict";
 
 const childProcess = require("child_process");
+const fs = require("fs");
 const assert = require("chai").assert;
 const EXECUTABLE_PATH = require("path").resolve(`${__dirname}/../../bin/eslint.js`);
 
 /**
+* Returns a Promise for when a child process exits
+* @param {ChildProcess} exitingProcess The child process
+* @returns {Promise} A Promise that fulfills when the child process exits
+*/
+function awaitExit(exitingProcess) {
```

Comment:
`assertExitCode` uses this same logic as part of its implementation. Is it worth changing it to call `awaitExit`? `awaitExit` would have to be modified to resolve with the exit code.

```js
function awaitExit(exitingProcess) {
    return new Promise(resolve => exitingProcess.once("exit", resolve));
}
// ...
function assertExitCode(exitingProcess, expectedExitCode) {
    return awaitExit(exitingProcess).then(exitCode => {
        if (exitCode !== expectedExitCode) {
            throw new Error(`Expected an exit code of ${expectedExitCode} but got ${exitCode}.`);
        }
    });
}
```